src/EditorFeatures/TestUtilities/EditAndContinue/MockEditAndContinueWorkspaceService.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable 
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.EditAndContinue.UnitTests
+{
+    internal class MockEditAndContinueWorkspaceService : IEditAndContinueWorkspaceService
+    {
+        [ExportWorkspaceServiceFactory(typeof(IEditAndContinueWorkspaceService), WorkspaceKind.Test), Shared]
+        internal sealed class Factory : IWorkspaceServiceFactory
+        {
+            [ImportingConstructor]
+            [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+            public Factory()
+            {
+            }
+
+            public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
+                => new MockEditAndContinueWorkspaceService();
+        }
+
+        public Func<ImmutableArray<DocumentId>, ImmutableArray<ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>>>? GetBaseActiveStatementSpansAsyncImpl;
+        public Func<Solution, ActiveInstructionId, LinePositionSpan?>? GetCurrentActiveStatementPositionAsyncImpl;
+        public Func<Document, ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>>? GetDocumentActiveStatementSpansAsyncImpl;
```

Comment:
cute :)    why not just have these be private though and have teh actual real interface methods call it?  seems just as simple and not as strange?

Reply:
I don't understand the suggestion. The point here is to make it easy to customize within a test method without defining a new subclass (similar to Moq).